### DISCUSSION TITLE
Symfony with Inky - How can we load the CSS to have working media queries?

### DISCUSSION THREAD

**Discussion by TristanPouliquen** [symfony/symfony#44923] [Q&A] [answered]
Hello,

We are trying to revamp our emails using the Inky helpers to make new more modern emails.

We have followed https://symfony.com/doc/current/mailer.html#inky-email-templating-language

However, we realise that the documentation forces the inline CSS method, which removes all media queries defined in the Foundation for Emails CSS.

What would be the best way of handling this so that we can still profit from the media queries (it allows to show/hide blocks which is pretty important in responsive design :-) )

**Comment by stof** [comment, marked as answer]
Well, the library used to implement the `inline_css` supports 2 ways to provide the CSS (that can actually be mixed):

- passing some CSS as argument, that will be used to apply inline rules (and rules that cannot be inlined are indeed ignored)
- putting some `<style>` tag in the HTML. The CSS in those tags is parsed to inline it. But the `<style>` tag also stays in the CSS, meaning that email clients that support style tags will still get the media queries

Using the second option would look like that:

```twig
{% apply inline_css %}
  <style>{{ source('@styles/foundation-emails.css') }}</style>
  {% apply inky_to_html %}
    <container>
        <row class="header">
            <columns>
                <spacer size="16"></spacer>
                <h1 class="text-center">Welcome {{ email.toName }}!</h1>
            </columns>

            {# ... #}
        </row>
    </container>
  {% endapply %}
{% endapply %}
```

You might maybe even use a single `apply` block using `inky_to_html|inline_css` if the `<style>` tag can be inside the content processed by Inky without breaking it (I think it is fine, but I'm not 100% sure).

And finally, you might also decide to rely only on the style tag instead of inlining CSS, but that might reduce the compatibility with email clients.

In any case, the Inky integration does not force using inlined CSS without media queries.

**Reply by stof** [reply]
btw, the documentations shows the different ways to use the `inline_css` filter: https://symfony.com/doc/current/mailer.html#inlining-css-styles (that's the same page, but scrolling up a few sections)

**Reply by TristanPouliquen** [reply]
Seems to be working with the <styles> tag indeed, thank you!

**Reply by mdieudonne** [reply]
On my side, for information, I have a good result with this syntax:

```
{% apply inline_css %}
    <style>
        {{ source('@styles/foundation-emails.css') }}
        {{ source('@styles/email.css') }}
    </style>
    {% apply inky_to_html %}
      your code goes here...
    {% endapply %}
{% endapply %}
```

But not with:
```
{% apply inline_css(source('@styles/foundation-emails.css'), source('@styles/email.css')) %}
    {% apply inky_to_html %}
    {% endapply %}
{% endapply %}
```
The dedicated style tag is a requirement

**Reply by stof** [reply]
That's because foundations-emails.css contains media queries, which cannot be inlined (this is not a limitation of the inliner, but of HTML itself). When using a style tag, this tag is preserved by the inlining, so email clients that support style tags will have the full styles with media queries. When passing the CSS directly to the inliner without having it in the HTML, those media queries are lost.

**Comment by yagami271** [comment]
Hello,
i test this solution to handle dark mode on email with 
`@media (prefers-color-scheme: dark) {....}` 

and i think it's not work. some idea ?
